src/main/java/commands/GoChapterCommand.java

```diff
@@ -31,32 +38,29 @@ public void execute(Ui ui, Access access, Storage storage) {
     }
 
     private String goChapter(Access access, Storage storage) {
-        boolean isLevelExist = false;
+        assert access.isModuleLevel() : "Not module level";
         String result = "";
-        String currentChapter = this.chapter;
-        ChapterList chapters = access.getModule().getChapters();
-        ArrayList<Chapter> allChapters = chapters.getAllChapters();
-        for (Chapter chapter : allChapters) {
-            String chapterName = chapter.getChapterName();
-            if (currentChapter.equalsIgnoreCase(chapterName)) {
-                access.setChapterLevel(chapterName);
-                isLevelExist = true;
-                try {
-                    ArrayList<Card> allCards = storage.loadCard(access.getModuleLevel(), chapter.getChapterName());
-                    if (allCards.size() == 0) {
-                        result = "This is a new chapter, you can try to add flashcards inside!";
-                    }
-                    chapter.setCards(allCards);
-                    access.setChapter(chapter);
-                } catch (FileNotFoundException e) {
-                    result = "The chapter file cannot be found.";
-                }
-                break;
+        try {
+            ChapterList chapters = access.getModule().getChapters();
+            ArrayList<Chapter> allChapters = chapters.getAllChapters();
+            Chapter chapter = allChapters.get(chapterIndex);
+            logger.info("Going into chapter: " + chapter.toString());
+            access.setChapterLevel(chapter.getChapterName());
```

Comment:
this line can be shift to after line 54: access.setChapter(chapter)
so you load the cards first. and if the chapter file cannot be found then access wont set anything yet
so access.setChapterLevel should be after loading the cards